addons/stock/models/stock_picking.py

```diff
@@ -728,6 +743,15 @@ def _action_done(self):
                     todo_moves |= new_move
         todo_moves._action_done(cancel_backorder=self.env.context.get('cancel_backorder'))
         self.write({'date_done': fields.Datetime.now()})
+
+        # if incoming moves make other confirmed/partially_available moves available, assign them
+        done_incoming_moves = self.filtered(lambda p: p.picking_type_id.code == 'incoming').mapped('move_lines').filtered(lambda m: m.state == 'done')
+        domains = []
+        for move in done_incoming_moves:
+            domains.append([('product_id', '=', move.product_id.id), ('location_id', '=', move.location_dest_id.id)])
+        moves_to_reserve = self.env['stock.move'].search(expression.AND([[('state', 'in', ['confirmed', 'partially_available']), ('procure_method', '=', 'make_to_stock')], expression.OR(domains)]))
```

Comment:
sorry to be picky but could you extract the static part of the domain to a variable so that it's 1) clearer 2) the line is shorter ?